### PR TITLE
feat(pipelines): allow user actions via popovers

### PR DESCRIPTION
**Pull request by anotherchrisberry**
A couple of structural changes:
1. Stages can provide a custom _icon_ for the execution marker.
2. Stages can now provide a custom _tooltip_, not just a custom _label_, that will be applied to execution markers.

This lets us do some sort of interesting things:
1. We can add icons to stages that are in a state that may be a little unintuitive. Stages that are awaiting manual judgment show an icon, as do stages that are waiting for an execution window to open:
screen shot 2017-04-22 at 12 15 15 pm

2. For stages that involves user interaction, we can let users do those things without having to click on the stage and get into its details. A simple manual judgment stage, when active and being hovered over:
screen shot 2017-04-21 at 10 43 28 pm
With instructions:
screen shot 2017-04-21 at 10 43 22 pm

A wait stage can be skipped over when it's running:
screen shot 2017-04-21 at 10 43 12 pm

Same with execution windows:
screen shot 2017-04-22 at 12 14 02 pm

From a React/technical side, I'm still figuring out React so please pick this apart as much as possible. One really nice thing about React is it is much easier to make a generic "hoverable popover" widget, something I never really figure out how to do cleanly with Angular.

### CODE REVIEW

File: app/scripts/modules/core/delivery/details/executionDetails.less [hunk 1]
```diff
@@ -1,7 +1,7 @@
 @import "../../presentation/less/imports/commonImports.less";
 
 .execution {
-  .execution-details {
+  execution-details, .execution-details {
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/delivery/details/executionDetails.less]: @jrsquared need both since `<execution-details>` is still a thing

File: app/scripts/modules/core/delivery/details/executionDetails.less [hunk 2]
```diff
@@ -159,3 +159,24 @@
...
+  .action-buttons {
+    display: flex;
+    align-items: stretch;
+    margin: 10px -5px 0;
+    button {
+      flex: 1 0 auto;
+      margin: 0 5px;
+    }
+  }
+}
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/delivery/details/executionDetails.less]: This change just lets buttons take up the normal amount of space in the details view, while filling the width of the popover.

File: app/scripts/modules/core/delivery/executionGroup/execution/Execution.tsx [hunk 3]
```diff
@@ -253,9 +253,12 @@ export class Execution extends React.Component<IExecutionProps, IExecutionState>
     const executionMarkerWidth = `${100 / this.props.execution.stageSummaries.length}%`;
     const executionMarkers = this.props.execution.stageSummaries.map((stage) => (
       <ExecutionMarker key={stage.refId}
+                       application={this.props.application}
+                       execution={this.props.execution}
                        stage={stage}
                        onClick={this.toggleDetails}
                        active={this.isActive(stage.index)}
+                       previousStageActive={this.isActive(stage.index - 1)}
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/delivery/executionGroup/execution/Execution.tsx]: We were doing some CSS business to tweak the execution bar borders, but it was relying on `+` selectors, which breaks down with the changes to `ExecutionMarker`. This is sort of better, anyway?

File: app/scripts/modules/core/pipeline/config/stages/manualJudgment/ManualJudgmentApproval.tsx [hunk 4]
```diff
@@ -61,51 +63,64 @@ export class ManualJudgmentApproval extends React.Component<IProps, IState> {
 
   private handleStopClick = () => this.provideJudgment('stop');
 
+  private getInstructions = (): any => {
+    return {
+      __html: DOMPurify.sanitize(this.props.stage.context.instructions)
+    };
+  };
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/pipeline/config/stages/manualJudgment/ManualJudgmentApproval.tsx]: Note: this is what the internet said is the way to safely render ad-hoc user-generated HTML.

File: app/scripts/modules/core/pipeline/config/stages/manualJudgment/manualJudgmentExecutionDetails.less [hunk 5]
```diff
@@ -3,6 +3,7 @@
     &.stage-type-manualjudgment {
       background-color: #F0AD4E;
       fill: #F0AD4E;
+      animation: none;
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/pipeline/config/stages/manualJudgment/manualJudgmentExecutionDetails.less]: no more blinking - we have an icon now

File: app/scripts/modules/core/pipeline/config/stages/wait/SkipWait.tsx [hunk 6]
```diff
@@ -15,7 +14,6 @@ interface IProps {
 }
 
 interface IState {
-  submitting: boolean;
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/pipeline/config/stages/wait/SkipWait.tsx]: Getting rid of the `submitting` - we were getting into an issue where the component would unmount and the call to `setState` below would complain. Really, we don't need it, and it wasn't there before. I was trying to be too fancy when I refactored this yesterday.

File: app/scripts/modules/core/delivery/executionGroup/execution/Execution.tsx (hunk 3 above)
Comment (jrsquared) [inline app/scripts/modules/core/delivery/executionGroup/execution/Execution.tsx]: Yes! I like this much better.

File: app/scripts/modules/core/pipeline/config/stages/manualJudgment/ManualJudgmentApproval.tsx (hunk 4 above)
Comment (jrsquared) [inline app/scripts/modules/core/pipeline/config/stages/manualJudgment/ManualJudgmentApproval.tsx]: Haha, yeah, super ugly. They really don't want us doing that.

File: app/scripts/modules/core/delivery/details/executionDetails.less (hunk 1 above)
Comment (jrsquared) [inline app/scripts/modules/core/delivery/details/executionDetails.less]: Only in 72 places, what's the problem?

File: app/scripts/modules/core/delivery/executionGroup/execution/ExecutionMarker.tsx [hunk 7]
```diff
@@ -1,16 +1,21 @@
 import * as React from 'react';
 import * as ReactGA from 'react-ga';
 
-import { IExecutionStageSummary } from 'core/domain/IExecutionStage';
+import { IExecution, IExecutionStageSummary } from 'core/domain';
 import { OrchestratedItemRunningTime } from './OrchestratedItemRunningTime';
-import { Tooltip } from 'core/presentation/Tooltip';
 import { duration } from 'core/utils/timeFormatters';
```
Comment (jrsquared) [inline app/scripts/modules/core/delivery/executionGroup/execution/ExecutionMarker.tsx]: whitespace between imports. also mixed `import { foo }` and `import {foo}`. I think I'm the only one left who prefers spaced imports for readability, I'm happy to convert to `import {foo}` since it's how all your inferior IDEs prefer it.

File: app/scripts/modules/core/delivery/executionGroup/execution/ExecutionMarker.tsx [hunk 8]
```diff
@@ -45,28 +50,40 @@ export class ExecutionMarker extends React.Component<IExecutionMarkerProps, IExe
   private handleStageClick = () => {
```
Comment (jrsquared) [inline app/scripts/modules/core/delivery/executionGroup/execution/ExecutionMarker.tsx]: If we merge @christopherthielen PR with the `@autobindMethods` decorator first, this should be updated.

File: app/scripts/modules/core/delivery/executionGroup/execution/ExecutionMarker.tsx [hunk 9]
```diff
@@ -45,28 +50,40 @@ export class ExecutionMarker extends React.Component<IExecutionMarkerProps, IExe
...
-    const stage = this.props.stage;
+    const {stage, application, execution} = this.props;
     const markerClassName = [
       'clickable',
       'stage',
       'execution-marker',
       `stage-type-${stage.type.toLowerCase()}`,
       `execution-marker-${stage.status.toLowerCase()}`,
       this.props.active ? 'active' : '',
+      this.props.previousStageActive ? 'after-active' : '',
```
Comment (jrsquared) [inline app/scripts/modules/core/delivery/executionGroup/execution/ExecutionMarker.tsx]: If you're going through the trouble of decomposing props, decompose `previousStageActive` and `active` too.

File: app/scripts/modules/core/delivery/executionGroup/execution/ExecutionMarker.tsx [hunk 10]
```diff
@@ -45,28 +50,40 @@ export class ExecutionMarker extends React.Component<IExecutionMarkerProps, IExe
...
       stage.isRunning ? 'glowing' : ''
       ].join(' ');
 
     const TooltipTemplate = stage.labelTemplate;
```
Comment (jrsquared) [inline app/scripts/modules/core/delivery/executionGroup/execution/ExecutionMarker.tsx]: Since you're in here... @christopherthielen made a great point that these are no longer `*Template` since I converted them to React. They should be `*Component`.

File: app/scripts/modules/core/delivery/executionGroup/execution/ExecutionMarker.tsx [hunk 11]
```diff
@@ -45,28 +50,40 @@ export class ExecutionMarker extends React.Component<IExecutionMarkerProps, IExe
...
+    const MarkerIcon = stage.markerIcon;
+    const stageContents = (
+      <div className={markerClassName}
+           style={{width: this.props.width, backgroundColor: stage.color}}
```
Comment (jrsquared) [inline app/scripts/modules/core/delivery/executionGroup/execution/ExecutionMarker.tsx]: decompose `width`

File: app/scripts/modules/core/delivery/executionGroup/execution/ExecutionMarker.tsx [hunk 12]
```diff
@@ -45,28 +50,40 @@ export class ExecutionMarker extends React.Component<IExecutionMarkerProps, IExe
...
+           onClick={this.handleStageClick}>
+        <MarkerIcon stage={stage}/>
+        <span className="duration">{duration(stage.runningTimeInMs)}</span>
```
Comment (jrsquared) [inline app/scripts/modules/core/delivery/executionGroup/execution/ExecutionMarker.tsx]: Just a "something to remember". I do this too, but the "React way" would be to do this `duration()` call somewhere else and set state with the duration (if you want it to rerender every time `stage.runningTimeInMs` changes.

File: app/scripts/modules/core/delivery/executionGroup/execution/execution.less [hunk 13]
```diff
@@ -94,10 +94,15 @@ execution-status {
   width: calc(~"100% - 235px");
   vertical-align: top;
   .stage {
+    .fa {
```
Comment (jrsquared) [inline app/scripts/modules/core/delivery/executionGroup/execution/execution.less]: can we rename this class? I have no idea what `fa` is...

File: app/scripts/modules/core/domain/IExecutionStage.ts [hunk 14]
```diff
@@ -22,9 +24,13 @@ export interface IExecutionStage extends IOrchestratedItem, IStage {
 }
 
 export interface IExecutionStageSummary extends IExecutionStage {
-  masterStage: IStage;
-  labelTemplate: React.ComponentClass<{ stage: IExecutionStageSummary }>;
+  masterStage: IExecutionStage;
+  stages: IExecutionStage[];
+  labelTemplate: React.ComponentClass<{ stage: IExecutionStageSummary, application?: Application, execution?: IExecution, executionMarker?: boolean }>;
```
Comment (jrsquared) [inline app/scripts/modules/core/domain/IExecutionStage.ts]: This typing makes me happy...

Comment (jrsquared) [inline app/scripts/modules/core/domain/IExecutionStage.ts]: Back to my previous comment that these should be labelComponent instead...

File: app/scripts/modules/core/pipeline/config/stages/core/ExecutionBarLabel.tsx [hunk 15]
```diff
@@ -1,12 +1,52 @@
 import * as React from 'react';
-import {IExecutionStageSummary} from 'core/domain/index';
+import {IExecutionStageSummary} from 'core/domain';
+import {ExecutionWindowActions} from 'core/pipeline/config/stages/executionWindows/ExecutionWindowActions';
+import {Application} from 'core/application/application.model';
+import {IExecution} from 'core/domain/IExecution';
```
Comment (jrsquared) [inline app/scripts/modules/core/pipeline/config/stages/core/ExecutionBarLabel.tsx]: Combine with prior `core/domain` import

File: app/scripts/modules/core/pipeline/config/stages/core/ExecutionBarLabel.tsx [hunk 16]
```diff
@@ -1,12 +1,52 @@
...
+          <ExecutionWindowActions application={application} execution={execution} stage={executionWindowStage}/>
+        </div>
+      );
+      return (
+        <HoverablePopover template={template}>
+          {this.props.children}
+        </HoverablePopover>
+      );
+    }
+    const tooltip = (
```
Comment (jrsquared) [inline app/scripts/modules/core/pipeline/config/stages/core/ExecutionBarLabel.tsx]: `const tooltip` should go inside the `if (executionMarker) {` block.

File: app/scripts/modules/core/pipeline/config/stages/core/ExecutionMarkerIcon.tsx [hunk 17]
```diff
@@ -0,0 +1,20 @@
+import * as React from 'react';
+import {IExecutionStageSummary} from 'core/domain/IExecutionStage';
+
+export interface IExecutionMarkerIconProps {
+  stage: IExecutionStageSummary;
+}
+
+export class ExecutionMarkerIcon extends React.Component<IExecutionMarkerIconProps, any> {
+
+  constructor(props: IExecutionMarkerIconProps) {
```
Comment (jrsquared) [inline app/scripts/modules/core/pipeline/config/stages/core/ExecutionMarkerIcon.tsx]: If the constructor is doing nothing, just omit it.

File: app/scripts/modules/core/pipeline/config/stages/core/ExecutionMarkerIcon.tsx [hunk 18]
```diff
@@ -0,0 +1,20 @@
...
+    super(props);
+  }
+
+  public render() {
+    if (this.props.stage.inSuspendedExecutionWindow) {
```
Comment (jrsquared) [inline app/scripts/modules/core/pipeline/config/stages/core/ExecutionMarkerIcon.tsx]: this could just be written `return this.props.stage.inSuspendedExecutionWindow ? <span className="fa fa-clock-o"/> : null;`

render() returns a `React.ReactElement< ExecutionMarkerIcon>` type, but every time I type that out, I think "why do we care?" since we never rely on the return of the render function directly. I'm happy to leave typings off of the render function...

File: app/scripts/modules/core/pipeline/config/stages/manualJudgment/ManualJudgmentExecutionLabel.tsx [hunk 19]
```diff
@@ -0,0 +1,54 @@
+import * as React from 'react';
+import {OverlayTrigger, Tooltip} from 'react-bootstrap';
+
+import {ManualJudgmentApproval} from './ManualJudgmentApproval';
+import {IExecutionStageSummary} from 'core/domain/index';
+import {IExecution} from 'core/domain/IExecution';
```
Comment (jrsquared) [inline app/scripts/modules/core/pipeline/config/stages/manualJudgment/ManualJudgmentExecutionLabel.tsx]: combine `core/domain`

File: app/scripts/modules/core/pipeline/config/stages/manualJudgment/ManualJudgmentExecutionLabel.tsx [hunk 20]
```diff
@@ -0,0 +1,54 @@
...
+import {Application} from 'core/application/application.model';
+import {HoverablePopover} from 'core/presentation/HoverablePopover';
+import {ExecutionBarLabel} from 'core/pipeline/config/stages/core/ExecutionBarLabel';
+
+interface IState {
+  target?: any;
+}
+
+interface IProps {
```
Comment (jrsquared) [inline app/scripts/modules/core/pipeline/config/stages/manualJudgment/ManualJudgmentExecutionLabel.tsx]: *Super* pedantic but since props comes before state in all functions (and logically as the component gets constructed, I like the `IProps` interface before the `IState` interface.

File: app/scripts/modules/core/pipeline/config/stages/manualJudgment/manualJudgmentStage.js [hunk 21]
```diff
@@ -15,6 +17,9 @@ module.exports = angular.module('spinnaker.core.pipeline.stage.manualJudgmentSta
       controllerAs: 'manualJudgmentStageCtrl',
       templateUrl: require('./manualJudgmentStage.html'),
       executionDetailsUrl: require('./manualJudgmentExecutionDetails.html'),
+      executionLabelTemplate: ManualJudgmentExecutionLabel,
```
Comment (jrsquared) [inline app/scripts/modules/core/pipeline/config/stages/manualJudgment/manualJudgmentStage.js]: `executionLabelComponent`

File: app/scripts/modules/core/pipeline/config/stages/wait/SkipWait.tsx [hunk 22]
```diff
@@ -46,17 +46,12 @@ export class SkipWait extends React.Component<IProps, IState> {
...
       body: '<p>The pipeline will proceed immediately, marking this stage completed.</p>',
       submitMethod: () => {
-        this.setState({submitting: true});
         return executionService.patchExecution(this.props.execution.id, stage.id, data)
           .then(() => executionService.waitUntilExecutionMatches(this.props.execution.id, matcher));
       }
     });
   };
 
-  public componentDidMount() {
```
Comment (jrsquared) [inline app/scripts/modules/core/pipeline/config/stages/wait/SkipWait.tsx]: nice.

File: app/scripts/modules/core/pipeline/config/stages/wait/WaitExecutionLabel.tsx [hunk 23]
```diff
@@ -0,0 +1,54 @@
+import * as React from 'react';
+import {OverlayTrigger, Tooltip} from 'react-bootstrap';
+
+import {IExecutionStageSummary} from 'core/domain/index';
+import {IExecution} from 'core/domain/IExecution';
```
Comment (jrsquared) [inline app/scripts/modules/core/pipeline/config/stages/wait/WaitExecutionLabel.tsx]: combine `core/domain`

File: app/scripts/modules/core/pipeline/config/stages/wait/WaitExecutionLabel.tsx [hunk 24]
```diff
@@ -0,0 +1,54 @@
...
+import {Application} from 'core/application/application.model';
+import {HoverablePopover} from 'core/presentation/HoverablePopover';
+import {ExecutionBarLabel} from 'core/pipeline/config/stages/core/ExecutionBarLabel';
+import {SkipWait} from './SkipWait';
+
+interface IState {
+  target?: any;
+}
+
+interface IProps {
```
Comment (jrsquared) [inline app/scripts/modules/core/pipeline/config/stages/wait/WaitExecutionLabel.tsx]: switch `IProps/IState`

File: app/scripts/modules/core/pipeline/config/stages/wait/WaitExecutionLabel.tsx [hunk 25]
```diff
@@ -0,0 +1,54 @@
...
+    super(props);
+    this.state = {};
+  }
+
+  public render() {
+    if (!this.props.executionMarker) {
+      return (<ExecutionBarLabel {...this.props}/>);
+    }
+    const stage = this.props.stage;
+    if (stage.isRunning) {
```
Comment (jrsquared) [inline app/scripts/modules/core/pipeline/config/stages/wait/WaitExecutionLabel.tsx]: This pattern of "If some case return `<HoverablePopover>`, else return `<OverlayTrigger>`" is repeated in a lot of these labels. Is there a way this could be pulled out into a different Component?

File: app/scripts/modules/core/presentation/HoverablePopover.tsx [hunk 26]
```diff
@@ -0,0 +1,69 @@
...
+import {Overlay, Popover} from 'react-bootstrap';
+import {UUIDGenerator} from '../utils/uuid.service';
+
+interface IState {
+  popoverIsOpen: boolean;
+  target?: any;
+  popoverCancel?: number;
+}
+
+interface IProps {
```
Comment (jrsquared) [inline app/scripts/modules/core/presentation/HoverablePopover.tsx]: switch

File: app/scripts/modules/core/presentation/HoverablePopover.tsx [hunk 27]
```diff
@@ -0,0 +1,69 @@
...
+    this.state = {
+      popoverIsOpen: false,
+    };
+  }
+
+  public componentWillUnmount() {
+    this.hidePopover();
+  }
+
+  private showPopover = (e: any): void => {
```
Comment (jrsquared) [inline app/scripts/modules/core/presentation/HoverablePopover.tsx]: `e` should have a type... something like `React.MouseEvent<HTMLElement>`

File: app/scripts/modules/core/presentation/HoverablePopover.tsx [hunk 28]
```diff
@@ -0,0 +1,69 @@
...
+
+  constructor(props: any) {
+    super(props);
+    this.state = {
+      popoverIsOpen: false,
+    };
+  }
+
+  public componentWillUnmount() {
+    this.hidePopover();
```
Comment (jrsquared) [inline app/scripts/modules/core/presentation/HoverablePopover.tsx]: this will probably throw up because there's a `setState` in `hidePopover()`. Should just call `this.clearPopoverCancel();` Wait, no can't do that, there's a setState in there too.

Now that I look more closely, `popoverCancel` probably shouldn't be on state. Should just be an instance variable, like the rx Subscriptions elsewhere in the code.

File: app/scripts/modules/core/presentation/HoverablePopover.tsx [hunk 29]
```diff
@@ -0,0 +1,69 @@
...
+  id?: string;
+}
+
+export class HoverablePopover extends React.Component<IProps, IState> {
+  public static defaultProps: Partial<IProps> = {
+    placement: 'top',
+    id: UUIDGenerator.generateUuid()
+  };
+
+  constructor(props: any) {
```
Comment (jrsquared) [inline app/scripts/modules/core/presentation/HoverablePopover.tsx]: `props: IProps`

File: app/scripts/modules/core/projects/dashboard/pipeline/ProjectPipeline.tsx [hunk 30]
```diff
@@ -8,8 +8,10 @@ import { $state } from 'core/uirouter';
 import { timestamp } from 'core/utils/timeFormatters';
 
 import './projectPipeline.less';
+import {Application} from 'core/application/application.model';
```
Comment (jrsquared) [inline app/scripts/modules/core/projects/dashboard/pipeline/ProjectPipeline.tsx]: Move this bad boy up with his friends.

File: app/scripts/modules/core/projects/dashboard/pipeline/ProjectPipeline.tsx [hunk 31]
```diff
@@ -34,7 +36,10 @@ export class ProjectPipeline extends React.Component<IProjectPipelineProps, IPro
   private handleStageClick = (stageIndex: number) => $state.go('^.application.pipelines.executionDetails.execution', {application: this.props.execution.application, executionId: this.props.execution.id, stage: stageIndex})
 
   public render() {
-    const stages = this.props.execution.stageSummaries.map((stage) => <ExecutionMarker key={stage.refId} stage={stage} onClick={this.handleStageClick} width={this.state.stageWidth}/>);
+    const application = this.props.application,
+          execution = this.props.execution,
+          stages = execution.stageSummaries.map((stage) =>
+      <ExecutionMarker key={stage.refId} application={application} execution={execution} stage={stage} onClick={this.handleStageClick} width={this.state.stageWidth}/>);
```
Comment (jrsquared) [inline app/scripts/modules/core/projects/dashboard/pipeline/ProjectPipeline.tsx]: A little JSX + ES6 magic if you want... since IProps is only application and execution this could change from:
```
   const application = this.props.application,
          execution = this.props.execution,
          stages = execution.stageSummaries.map((stage) =>
   <ExecutionMarker key={stage.refId} application={application} execution={execution} stage={stage} onClick={this.handleStageClick} width={this.state.stageWidth}/>);
```

to:
```
const stages = execution.stageSummaries.map((stage) =>
   <ExecutionMarker key={stage.refId} {...this.props} stage={stage} onClick={this.handleStageClick} width={this.state.stageWidth}/>);

File: app/scripts/modules/core/pipeline/config/stages/manualJudgment/ManualJudgmentApproval.tsx (hunk 4 above)
Comment (anotherchrisberry) [inline app/scripts/modules/core/pipeline/config/stages/manualJudgment/ManualJudgmentApproval.tsx]: Unfortunately, it won't be the last time we have to do this. Probably worth refactoring into a common module next time we do it.

File: app/scripts/modules/core/delivery/executionGroup/execution/ExecutionMarker.tsx (hunk 7 above)
Comment (anotherchrisberry) [inline app/scripts/modules/core/delivery/executionGroup/execution/ExecutionMarker.tsx]: I'm good with the spaces and have updated my IDE to include them in import statements.

File: app/scripts/modules/core/delivery/executionGroup/execution/ExecutionMarker.tsx (hunk 12 above)
Comment (anotherchrisberry) [inline app/scripts/modules/core/delivery/executionGroup/execution/ExecutionMarker.tsx]: good call, will move the `duration` invocation to `setState`

File: app/scripts/modules/core/delivery/executionGroup/execution/execution.less (hunk 13 above)
Comment (anotherchrisberry) [inline app/scripts/modules/core/delivery/executionGroup/execution/execution.less]: it's "font-awesome" - will add a comment to the less file explaining that

File: app/scripts/modules/core/domain/IExecutionStage.ts (hunk 14 above)
Comment (anotherchrisberry) [inline app/scripts/modules/core/domain/IExecutionStage.ts]: Generally agreed, but will do this in a separate PR once this is merged for clarity

File: app/scripts/modules/core/pipeline/config/stages/core/ExecutionBarLabel.tsx (hunk 16 above)
Comment (anotherchrisberry) [inline app/scripts/modules/core/pipeline/config/stages/core/ExecutionBarLabel.tsx]: Good catch.

File: app/scripts/modules/core/pipeline/config/stages/manualJudgment/ManualJudgmentExecutionLabel.tsx (hunk 20 above)
Comment (anotherchrisberry) [inline app/scripts/modules/core/pipeline/config/stages/manualJudgment/ManualJudgmentExecutionLabel.tsx]: No, I like that, too.

File: app/scripts/modules/core/pipeline/config/stages/manualJudgment/manualJudgmentStage.js (hunk 21 above)
Comment (anotherchrisberry) [inline app/scripts/modules/core/pipeline/config/stages/manualJudgment/manualJudgmentStage.js]: (separate PR)

File: app/scripts/modules/core/presentation/HoverablePopover.tsx (hunk 28 above)
Comment (anotherchrisberry) [inline app/scripts/modules/core/presentation/HoverablePopover.tsx]: good call

File: app/scripts/modules/core/pipeline/config/stages/wait/WaitExecutionLabel.tsx (hunk 25 above)
Comment (anotherchrisberry) [inline app/scripts/modules/core/pipeline/config/stages/wait/WaitExecutionLabel.tsx]: Maybe? There's just enough difference between the three places with similar logic that I'm not convinced there's a good abstraction.

File: app/scripts/modules/core/presentation/HoverablePopover.tsx (hunk 28 above)
Comment (anotherchrisberry) [inline app/scripts/modules/core/presentation/HoverablePopover.tsx]: (done)
